File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {

        int mini = prices[0];

        int maxi = 0;


        for(int i=0;i<prices.length;i++){
            int cost = prices[i] - mini;

            mini = Math.min(prices[i], mini);
            maxi = Math.max(maxi, cost);
        }

        return maxi;
        
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
var maxProfit = function (prices) {
  let n = prices.length;

  let max = 0;
  let currentLowest = prices[0];

  for (let i = 0; i < n; i++) {
    if (prices[i] < currentLowest) currentLowest = prices[i];

    max = Math.max(max, prices[i] - currentLowest);
  }

  return max;
};
